include/ck_tile/ops/gemm/kernel/gemm_tile_partitioner.hpp

```diff
@@ -1,73 +1,130 @@
+    // blockIdx is blockId.x and blockIdy is blockIdx.y.
+    CK_TILE_DEVICE static constexpr auto GetOutputTileIndex(index_t blockIdx,
+                                                            index_t blockIdy) noexcept
+        -> const tuple<index_t, index_t>
     {
-        const index_t iM = __builtin_amdgcn_readfirstlane(blockIdx.x * kM);
-        const index_t iN = __builtin_amdgcn_readfirstlane(blockIdx.y * kN);
+        const index_t iM = __builtin_amdgcn_readfirstlane(blockIdx);
+        const index_t iN = __builtin_amdgcn_readfirstlane(blockIdy);
         return make_tuple(iM, iN);
     }
 };
 
-template <typename BlockGemmShape_>
+/// @brief An internal ck-tile class dedicated to 1D tiles.
+template <typename BlockGemmShapeType>
 struct GemmTile1DPartitioner
 {
-    using BlockGemmShape = remove_cvref_t<BlockGemmShape_>;
+    using BlockGemmShape = remove_cvref_t<BlockGemmShapeType>;
 
     static constexpr index_t MPerBlock = BlockGemmShape::kM;
     static constexpr index_t NPerBlock = BlockGemmShape::kN;
     static constexpr index_t KPerBlock = BlockGemmShape::kK;
 
-    CK_TILE_HOST static constexpr auto GridSize(index_t M, index_t N)
+    CK_TILE_HOST static constexpr auto
+    GridSize(index_t M, index_t N) noexcept(noexcept(MPerBlock != 0 && NPerBlock != 0)) -> dim3
     {
-        index_t GridDimX = (M + MPerBlock - 1) / MPerBlock;
-        index_t GridDimY = (N + NPerBlock - 1) / NPerBlock;
+        const index_t GridDimX = (M + MPerBlock - 1) / MPerBlock;
+        const index_t GridDimY = (N + NPerBlock - 1) / NPerBlock;
         return dim3(GridDimX * GridDimY, 1, 1);
     }
 
-    CK_TILE_HOST_DEVICE static constexpr auto GetNBlock(index_t N)
+    CK_TILE_HOST_DEVICE static constexpr auto GetNBlock(index_t N) noexcept -> index_t
     {
         return integer_divide_ceil(N, NPerBlock);
     }
 
-    CK_TILE_HOST_DEVICE static constexpr auto GetLoopNum(index_t K)
+    CK_TILE_HOST_DEVICE static constexpr auto GetLoopNum(index_t K) noexcept -> index_t
     {
         return integer_divide_ceil(K, KPerBlock);
     }
 
-    CK_TILE_DEVICE auto operator()(index_t blockOffset, index_t NBlockSize)
+    CK_TILE_DEVICE static constexpr auto SetNBlock(index_t N) noexcept -> void { _NBlockSize = N; }
+
+    /// @brief Returns the output tile index. Call with blockIdx, which is offset (blockIdx.x -
+    /// block_start). Note: The `SetNBlock` must be called before calling this function.
+    CK_TILE_DEVICE static constexpr auto GetOutputTileIndex(index_t blockIdx) noexcept
+        -> const tuple<index_t, index_t>
+    {
+        const index_t NBlock = GetNBlock(_NBlockSize);
+
+        const index_t iM = __builtin_amdgcn_readfirstlane(blockIdx / NBlock);
+        const index_t iN = __builtin_amdgcn_readfirstlane(modulo(blockIdx, NBlock));
+        return make_tuple(iM, iN);
+    }
+
+    private:
+    CK_TILE_DEVICE static index_t _NBlockSize;
```

Comment:
Aren't `_` underscore prefixes reserved for `std::` or compiler internal variables? Let's use suffix rather. And this variable is not actually a `NBlockSize` but rather gemm's `N` dimension.